tracer/src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp

```diff
@@ -395,6 +417,46 @@ namespace datadog::shared::nativeloader
 
     HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID moduleId, HRESULT hrStatus)
     {
+        try
+        {
+            if (isWriteInstrumentationToDiskEnabled)
+            {
+                WCHAR modulePath[MAX_PATH];
+                DWORD nameSize = 0;
+                LPCBYTE baseLoadAddress;
+                AssemblyID assemblyId = 0;
+                DWORD moduleFlags = 0;
+                const auto hr = m_writeToDiskCorProfilerInfo->GetModuleInfo2(moduleId, &baseLoadAddress, MAX_PATH, &nameSize,
+                                                                  modulePath, &assemblyId, &moduleFlags);
+
+                if (FAILED(hr))
+                {
+                    Log::Warn("InstrumentationVerification: fail on call GetModuleInfo2 for moduleId {}", moduleId);
+                }
+                else
+                {
+                    ::shared::WSTRING moduleName;
+                    if (std::char_traits<WCHAR>::length(modulePath) == 0)
```

Comment:
in `string.h` you have a `WStrLen`.
in your case you can check the `nameSize` which will contain the number of characters in `modulePath`.